nautobot/dcim/tests/test_api.py

```diff
@@ -2410,3 +2443,130 @@ def setUpTestData(cls):
             "failover_strategy": "active-passive",
             "status": statuses[1].pk,
         }
+        # FIXME(jathan): The writable serializer for `status` takes the
+        # status `name` (str) and not the `pk` (int). Do not validate this
+        # field right now, since we are asserting that it does create correctly.
+        #
+        # The test code for `nautobot.core.testing.views.TestCase.model_to_dict()`
+        # needs to be enhanced to use the actual API serializers when `api=True`
+        cls.validation_excluded_fields = ["status"]
```

Comment:
Without this line, the unit tests do fail with the following errors.
```
======================================================================
FAIL: test_bulk_create_objects (nautobot.dcim.tests.test_api.InterfaceRedundancyGroupTestCase)
POST a set of objects in a single request.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/source/nautobot/core/testing/api.py", line 730, in test_bulk_create_objects
    self.assertInstanceEqual(
  File "/source/nautobot/core/testing/mixins.py", line 202, in assertInstanceEqual
    self.assertEqual(new_model_dict, relevant_data)
AssertionError: {'nam[32 chars]4', 'status': UUID('41889e54-e444-43d7-974e-e0[177 chars]9a')} != {'nam[32 chars]4', 'protocol': 'hsrp', 'status': 'Active', 'p[141 chars]9a')}
  {'name': 'Interface Redundancy Group 4',
   'protocol': 'hsrp',
   'protocol_group_id': '1',
   'secrets_group': UUID('13615a06-2333-4389-a271-10767c2ec84a'),
-  'status': UUID('41889e54-e444-43d7-974e-e00ce0b164ab'),
+  'status': 'Active',
   'virtual_ip': UUID('0106d4f0-9b26-4670-9cb2-a87ea631879a')}

======================================================================
FAIL: test_bulk_update_objects (nautobot.dcim.tests.test_api.InterfaceRedundancyGroupTestCase)
PATCH a set of objects in a single request.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/source/nautobot/core/testing/api.py", line 897, in test_bulk_update_objects
    self.assertInstanceEqual(
  File "/source/nautobot/core/testing/mixins.py", line 202, in assertInstanceEqual
    self.assertEqual(new_model_dict, relevant_data)
AssertionError: {'status': UUID('41889e54-e444-43d7-974e-e0[88 chars]9a')} != {'protocol': 'carp', 'status': 'Active', 'v[52 chars]9a')}
  {'protocol': 'carp',
-  'status': UUID('41889e54-e444-43d7-974e-e00ce0b164ab'),
+  'status': 'Active',
   'virtual_ip': UUID('0106d4f0-9b26-4670-9cb2-a87ea631879a')}

======================================================================
FAIL: test_create_object (nautobot.dcim.tests.test_api.InterfaceRedundancyGroupTestCase)
POST a single object with permission.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/source/nautobot/core/testing/api.py", line 635, in test_create_object
    self.assertInstanceEqual(
  File "/source/nautobot/core/testing/mixins.py", line 202, in assertInstanceEqual
    self.assertEqual(new_model_dict, relevant_data)
AssertionError: {'nam[32 chars]4', 'status': UUID('41889e54-e444-43d7-974e-e0[177 chars]9a')} != {'nam[32 chars]4', 'protocol': 'hsrp', 'status': 'Active', 'p[141 chars]9a')}
  {'name': 'Interface Redundancy Group 4',
   'protocol': 'hsrp',
   'protocol_group_id': '1',
   'secrets_group': UUID('13615a06-2333-4389-a271-10767c2ec84a'),
-  'status': UUID('41889e54-e444-43d7-974e-e00ce0b164ab'),
+  'status': 'Active',
   'virtual_ip': UUID('0106d4f0-9b26-4670-9cb2-a87ea631879a')}

======================================================================
FAIL: test_update_object (nautobot.dcim.tests.test_api.InterfaceRedundancyGroupTestCase)
PATCH a single object identified by its ID.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/source/nautobot/core/testing/api.py", line 831, in test_update_object
    self.assertInstanceEqual(instance, update_data, exclude=self.validation_excluded_fields, api=True)
  File "/source/nautobot/core/testing/mixins.py", line 202, in assertInstanceEqual
    self.assertEqual(new_model_dict, relevant_data)
AssertionError: {'nam[32 chars]4', 'status': UUID('41889e54-e444-43d7-974e-e0[177 chars]9a')} != {'nam[32 chars]4', 'protocol': 'hsrp', 'status': 'Active', 'p[141 chars]9a')}
  {'name': 'Interface Redundancy Group 4',
   'protocol': 'hsrp',
   'protocol_group_id': '1',
   'secrets_group': UUID('13615a06-2333-4389-a271-10767c2ec84a'),
-  'status': UUID('41889e54-e444-43d7-974e-e00ce0b164ab'),
+  'status': 'Active',
   'virtual_ip': UUID('0106d4f0-9b26-4670-9cb2-a87ea631879a')}
```

Reply:
Ah. That probably relates to my other comment about using `status.name` versus `status.pk` in the `create_data` - these tests naively expect the data passed into the API to match the data returned by the API, and since the return is going to be a PK, the create_data probably needs to also use a PK to make these tests happy.